extensions/skins/conversation_v1/Conversation.js

```diff
@@ -643,6 +646,25 @@ oppia.directive('progressDots', [function() {
           $scope.changeActiveDot($scope.currentDotIndex + 1);
         }
       };
+
+      /* This is needed to watch any change for both number of dots and currentDotIndex. */
+      $scope.$watchGroup(['dots', 'currentDotIndex'], function(oldValue, newValue) {
+        if(newValue !== oldValue) {
+          if($scope.dots.length > $scope.MAX_DOTS) {
+            if($scope.currentDotIndex < $scope.dots.length - $scope.MAX_DOTS) {
+              $scope.leftmostVisibleDotIndex = $scope.currentDotIndex;
```

Comment:
Done!.
Eg. i have created 22 state exploration
![last](https://cloud.githubusercontent.com/assets/6409210/10065970/ffb2668a-6295-11e5-9f8d-f32f41f9e80d.png)

but if you go to index 0(state 1). 
![0twent2](https://cloud.githubusercontent.com/assets/6409210/10065801/ed93fbc8-6293-11e5-81ec-4bf83c2ecb0b.png)
the range will stay the same (0-17 for this example case) untill ```currentDotPosition```  is at index that is greater than ```HALF_DOTS_RANGE```. snapshot below shows  11 state and you can see index zero does not fade and left arrow is active to navigate left as ussual 
![eleven](https://cloud.githubusercontent.com/assets/6409210/10065940/9a78ddee-6295-11e5-80bb-a84426614d34.png)

and if  ```currentDotPosition``` is at the position greater than ```HALF_DOTS_RANGE``` to the left and right fading effects will in both sides and ``currentDotPosition``` will be at the middle.
![3to20](https://cloud.githubusercontent.com/assets/6409210/10065883/f23c9364-6294-11e5-80da-6e41daedd099.png)
in the above means there are 3 dots hidden to the left(0-2) and 2 to the right(21,22).  I guess this was desired results?, if not I will be more than happy to make it better.